spec/support/auth_support.rb

```diff
@@ -1,9 +1,20 @@
 require 'rails_helper'
+
 module AuthSupport
   
   def stub_auth
     @request.env["devise.mapping"] = Devise.mappings[:user]
     user = FactoryBot.create(:user)
-    sign_in user
+    allow(request.env['warden']).to receive(:authenticate!).and_return(user)
+    allow(controller).to receive(:current_user).and_return(user)
+    sign_in(user, scope: :user)
+  end
+
+  def certmgr_stub_auth
+    @request.env["devise.mapping"] = Devise.mappings[:user]
+    user = FactoryBot.create(:component_user)
+    allow(request.env['warden']).to receive(:authenticate!).and_return(user)
+    allow(controller).to receive(:current_user).and_return(user)
+    sign_in(user, scope: :user)
```

Comment:
these 3 lines are repeated in each method, we could extract them to reduce some of the code duplication...